interactions/api/models/message.py

```diff
@@ -888,7 +888,7 @@ async def edit(
             _attachments = [a._json for a in attachments]
 
         if not files or files is MISSING:
-            _files = self.attachments
+            _files = [] if attachments is MISSING else [a._json for a in self.attachments]
```

Comment:
```suggestion
            _files = None
```

Reply:
1. If `attachments` is not missing?
2. Then just `_files = []` because you have below `_files.extend`